File: py/10.py
from math import sqrt

def prime(n):
    if n % 2 == 0:
        return False

    for i in range(3, int(sqrt(n)) + 1):
        if n != i and n % i == 0:
            return False

    return True

sum = 0

for i in xrange(3, 2000000, 2):
    if prime(i):
        sum += i

print sum + 2

File: hs/10.hs
fs = floor . sqrt . fromIntegral

prime n | n == 2    = True
        | even n    = False
        | otherwise = notElem 0 $ map (\m -> mod n m) [3,5..fs n]

main = print $ foldl1 (+) $ filter prime [2..2000000]
